ivory/src/Ivory/Language/Syntax/Concrete/QQ/BitDataQQ.hs

```diff
@@ -351,7 +353,11 @@ mkDefInstance def = [instanceD (cxt []) instTy body]
     baseTy  = thDefType def
     instTy  = [t| B.BitData $(conT (thDefName def)) |]
     body    = [tyDef, toFun, fromFun]
+#if MIN_VERSION_th_abstraction(0,3,0)
+    tyDef   = tySynInstDCompat ''B.BitType Nothing [conT name] (return baseTy)
+#else
     tyDef   = tySynInstDCompat ''B.BitType [conT name] (return baseTy)
+#endif
```

Comment:
Let's just bump the lower bound on `th-abstraction` instead of adding compatibility shims for compatibility shims.